kernel/dev/rtc/rtc.cpp

```diff
@@ -136,23 +139,41 @@ void read_rtc() {
     }
 
     // Calculate the full (4-digit) year
-    if(RTC_CURRENT_CENTURY != 0) {
+    if (RTC_CURRENT_CENTURY != 0) {
         rtc_year += rtc_century * 100;
     } else {
         rtc_year += (RTC_CURRENT_YEAR / 100) * 100;
-        if(rtc_year < RTC_CURRENT_YEAR) rtc_year += 100;
+        if (rtc_year < RTC_CURRENT_YEAR)
+            rtc_year += 100;
+    }
+}
+
+static uint64_t getJulianDay(uint8_t days, uint8_t months, uint16_t years)
+{
+    // If month is before march
+    if (months < 3) {
+        years -= 1;
+        months += 12;
     }
+    uint64_t a = years / 100;
+    uint64_t b = a / 4;
+    uint64_t c = 2 - a + b;
+    uint64_t d = 365.25 * (years + 4716);
```

Comment:
You can't do `double` (numbers with decimal points) math using integers. The decimal points will just get dropped.